### PR TITLE
Doc more release updates

### PR DESCRIPTION
**Pull request by tacaswell**
More updates to the release docs.

This is more documenting what is actually done rather than prescribing what should be done.

### CODE REVIEW

File: tools/github_stats.py [hunk 1]
```diff
@@ -227,3 +227,16 @@ def report(issues, show_urls=False):
         print()
         print('Issues (%d):\n' % n_issues)
         report(issues, show_urls)
+    print()
+    print()
+    print("""Previous Github Stats
```
Comment (QuLogic) [inline tools/github_stats.py]:
```suggestion
    print("""Previous GitHub Stats
```

File: doc/devel/release_guide.rst [hunk 2]
```diff
@@ -65,6 +74,9 @@ Review and commit changes.  Some issue/PR titles may not be valid rst (the most
 Update and Validate the Docs
 ----------------------------
 
+Merge the most recent 'doc' branch (``v3.0.2-doc``) into the branch you
+are going to tag on and delete the doc branch on github.
```
Comment (QuLogic) [inline doc/devel/release_guide.rst]:
```suggestion
are going to tag on and delete the doc branch on GitHub.
```

File: tools/github_stats.py [hunk 3]
```diff
@@ -227,3 +227,16 @@ def report(issues, show_urls=False):
...
+---------------------
+
+
+.. toctree::
+    :maxdepth: 1
+    :glob:
+
+    github_stats_*
+
+"""
```
Comment (timhoffm) [inline tools/github_stats.py]: Missing `)`.

Comment (tacaswell) [reply tools/github_stats.py]: fixed and force-pushed.  On one hand, I am glad the UI now shows force pushes for traceability, on the other hand it makes it harder to hide embarrassing mistakes like this ;)

**Pull request merged**